File: Python_Fundamental/Functions/perfect_number.py
def perfect_number(number_to_check):
    sum_of_dividers = 0
    for num in range(1, number):
        if number % num == 0:
            sum_of_dividers += num
    if sum_of_dividers == number_to_check:
        print("We have a perfect number!" )
    else:
        print("It's not so perfect.")


number = int(input())
perfect_number(number)

File: JavaScript/Functions_and_statements/perfect_number.js


function perfectNnumber(number) {
    let divisors = []
    let result = 0

    for (let i = 1; i < number; i++){
        if (number % i === 0){
            divisors.push(i)
        }
    }

    for (const i of divisors) {
        currentNumber = Number(i)
        result += currentNumber
    }

    if (result === number){
        console.log("We have a perfect number!")
    } else {
        console.log("It's not so perfect.")
    }
}


perfectNnumber(6)